prov/tcp/src/tcpx_progress.c

```diff
@@ -117,18 +117,72 @@ static void process_rx_pe_entry(struct tcpx_pe_entry *pe_entry)
 	tcpx_pe_entry_release(pe_entry);
 }
 
-static void process_rx_queue(struct tcpx_ep *ep)
+static struct tcpx_pe_entry * tcpx_get_rx_pe_entry(struct tcpx_rx_hdr *rx_hdr)
 {
-	struct tcpx_pe_entry *pe_entry;
+	struct tcpx_pe_entry *pe_entry = NULL;
 	struct dlist_entry *entry;
+	struct tcpx_ep *tcpx_ep;
+
+	tcpx_ep = container_of(rx_hdr, struct tcpx_ep, rx_hdr);
+
+	switch (rx_hdr->hdr.op) {
+	case ofi_op_msg:
+		if (dlist_empty(&tcpx_ep->rx_queue))
+			return NULL;
+
+		entry = tcpx_ep->rx_queue.next;
+		pe_entry = container_of(entry, struct tcpx_pe_entry,
+					entry);
+
+		pe_entry->msg_hdr = rx_hdr->hdr;
+		pe_entry->msg_hdr.op_data = TCPX_OP_MSG_RECV;
+		pe_entry->done_len = sizeof(rx_hdr->hdr);
+
+		if (ofi_truncate_iov(pe_entry->msg_data.iov,
+				     &pe_entry->msg_data.iov_cnt,
+				     (ntohll(pe_entry->msg_hdr.size) -
+				      sizeof(pe_entry->msg_hdr))))
```

Comment:
The purpose of calling this function isn't obvious to me.  Neither is error handling that occurs if it fails.